InformationScripting/src/queries/QueryExecutor.h

```diff
@@ -28,24 +28,34 @@
 
 #include "../informationscripting_api.h"
 
+#include "../misc/Optional.h"
+#include "TopLevelQuery.h"
+
 namespace Interaction {
 	class CommandResult;
 }
 
 namespace InformationScripting {
 
+class TopLevelQuery;
+class TupleSet;
 class Query;
 
 class INFORMATIONSCRIPTING_API QueryExecutor
 {
 	public:
-		QueryExecutor(Query* q);
 		~QueryExecutor();
+		void addQuery(std::unique_ptr<TopLevelQuery>&& query);
 
 		Interaction::CommandResult* execute();
 
+		QList<Optional<TupleSet> > runContinuation(const QList<TupleSet>& input);
```

Comment:
If we execute from the command prompt we would like to have a CommandResult if we execute a continuation we would like to have the return value from the query. Not sure how to solve this nicely. Even though I fully agree with your concern.